public/app/plugins/datasource/elasticsearch/components/QueryEditor/LuceneEditor.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { FunctionComponent } from 'react';
+import { ElasticsearchQuery, ElasticsearchQueryType } from '../../types';
+import { InlineField, InlineFieldRow, Input, QueryField } from '@grafana/ui';
+import { changeAliasPattern, changeQuery } from './state';
+import { QueryTypeEditor } from './QueryTypeEditor';
+import { MetricAggregationsEditor } from './MetricAggregationsEditor';
+import { BucketAggregationsEditor } from './BucketAggregationsEditor';
+import { useDispatch } from '../../hooks/useStatelessReducer';
+import { useNextId } from '../../hooks/useNextId';
+
+interface Props {
+  query: ElasticsearchQuery['query'];
+}
+
+export const LuceneEditor: FunctionComponent<Props> = ({ query }) => {
+  const dispatch = useDispatch();
+  const nextId = useNextId();
+
+  return (
+    <>
+      <InlineFieldRow>
+        <InlineField label="Query" labelWidth={17} grow>
+          <>
+            <QueryTypeEditor value={ElasticsearchQueryType.Lucene} />
+            <QueryField
+              query={query}
+              // By default QueryField calls onChange if onBlur is not defined, this will trigger a rerender
+              // And slate will claim the focus, making it impossible to leave the field.
+              onBlur={() => {}}
```

Comment:
does defining `handleBlur` work? https://grafana.com/docs/grafana/latest/packages_api/ui/queryfield/#handleblur-property

Reply:
I think this is the pattern they use in all instances of the `QueryField` in the codebase, basically we just need to pass a function into `onBlur` props but want it to be noop.